File: python/large-class/good/email_service.py
class EmailService:
    def __init__(self, email_config):
        self.smtp_username = email_config.get('username')

    def send_welcome_email(self, email: str, name: str) -> bool:
        subject = 'Welcome to our platform!'
        message = f"Hello {name},\n\nWelcome to our platform!"

        print(f"Sending email to {email} from {self.smtp_username}")
        print(f"Subject: {subject}")
        print(f"Message: {message}")
        return True

    def send_password_reset_email(self, email: str, reset_token: str) -> bool:
        subject = 'Password Reset'
        message = f"Click here to reset your password: http://example.com/reset?token={reset_token}"

        print(f"Sending email to {email} from {self.smtp_username}")
        print(f"Subject: {subject}")
        print(f"Message: {message}")
        return True

    def send_notification_email(self, email: str, subject: str, message: str) -> bool:
        print(f"Sending email to {email} from {self.smtp_username}")
        print(f"Subject: {subject}")
        print(f"Message: {message}")
        return True

File: php/large-class/good/EmailService.php
<?php

class EmailService {
    private string $smtpHost;
    private int $smtpPort;
    private string $smtpUsername;
    private string $smtpPassword;

    public function __construct(array $emailConfig) {
        $this->smtpHost = $emailConfig['host'];
        $this->smtpPort = $emailConfig['port'];
        $this->smtpUsername = $emailConfig['username'];
        $this->smtpPassword = $emailConfig['password'];
    }

    public function sendWelcomeEmail(string $email, string $name): bool {
        $subject = 'Welcome to our platform!';
        $message = "Hello {$name},\n\nWelcome to our platform!";

        $headers = 'From: ' . $this->smtpUsername . "\r\n";
        $headers .= 'Reply-To: ' . $this->smtpUsername . "\r\n";

        return mail($email, $subject, $message, $headers);
    }

    public function sendPasswordResetEmail(string $email, string $resetToken): bool {
        $subject = 'Password Reset';
        $message = "Click here to reset your password: http://example.com/reset?token={$resetToken}";

        $headers = 'From: ' . $this->smtpUsername . "\r\n";

        return mail($email, $subject, $message, $headers);
    }

    public function sendNotificationEmail(string $email, string $subject, string $message): bool {
        $headers = 'From: ' . $this->smtpUsername . "\r\n";

        return mail($email, $subject, $message, $headers);
    }
}
